src/Tracing/Tracing.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Nuwave\Lighthouse\Tracing;
+
+use Carbon\Carbon;
+use GraphQL\Type\Definition\ResolveInfo;
+use Nuwave\Lighthouse\Events\StartRequest;
+use Nuwave\Lighthouse\Schema\AST\ASTHelper;
+use Nuwave\Lighthouse\Events\StartExecution;
+use Nuwave\Lighthouse\Events\ManipulatingAST;
+use Nuwave\Lighthouse\Schema\AST\PartialParser;
+use Nuwave\Lighthouse\Events\GatheringExtensions;
+
+class Tracing
+{
+    /**
+     * @var \Carbon\Carbon
+     */
+    protected $requestStart;
+
+    /**
+     * Trace entries.
+     *
+     * @var array[]
+     */
+    protected $resolverTraces = [];
+
+    /**
+     * Set the tracing directive on all fields of the query to enable tracing them.
+     *
+     * @param  \Nuwave\Lighthouse\Events\ManipulatingAST  $manipulatingAST
+     * @return void
+     */
+    public function handleManipulatingAST(ManipulatingAST $manipulatingAST): void
```

Comment:
I would just call it `handle`

Reply:
As you scroll further down this class, you will see it handles many different events :)